### PR TITLE
fix #302304: prevent potential crash when adding instruments

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
esp. to score with parts imported from MuseScore 1.x

Resolves https://musescore.org/en/node/302304

### CODE REVIEW

File: libmscore/midimapping.cpp [hunk 1]
```diff
@@ -143,7 +143,7 @@ void MasterScore::rebuildExcerptsMidiMapping()
             for (Part* p : ex->partScore()->parts()) {
                   const Part* masterPart = p->masterPart();
                   if (!masterPart->score()->isMaster()) {
-                        qWarning("reorderMidiMapping: no part in master score is linked");
+                        qWarning() << QString("rebuildExcerptsMidiMapping: no part in master score is linked with %1").arg(p->partName());
```
Comment (dmitrio95) [inline libmscore/midimapping.cpp]: There is no need to use `.arg()` here, just successive `<<` operators would be enough:
```
qWarning() << "rebuildExcerptsMidiMapping: no part ..." << p->partName();
```

Comment (Jojo-Schmitz) [reply libmscore/midimapping.cpp]: I tried that, didn't work

Comment (Jojo-Schmitz) [reply libmscore/midimapping.cpp]: Looking again now...

Comment (Jojo-Schmitz) [reply libmscore/midimapping.cpp]: Apparently I didn't try that variant ;-)

File: mscore/instrdialog.cpp [hunk 2]
```diff
@@ -519,6 +519,8 @@ void MuseScore::editInstrList()
             else {
                   for (Staff* s : sl) {
                         const LinkedElements* sll = s->links();
+                        if (!sll)
+                              break;
```
Comment (dmitrio95) [inline mscore/instrdialog.cpp]: Why not `continue`? Then all staves which have links would be handled, and the current fix will stop at the first staff which doesn't have links.

Comment (Jojo-Schmitz) [reply mscore/instrdialog.cpp]: good question... let's to that

Comment (Jojo-Schmitz) [reply mscore/instrdialog.cpp]: done

**Pull request merged**